packages/integrations/src/pi-hole/pi-hole-integration.ts

```diff
@@ -0,0 +1,34 @@
+import { Integration } from "../base/integration";
+import type { DnsHoleSummaryIntegration } from "../interfaces/dns-hole-summary/dns-hole-summary-integration";
+import type { DnsHoleSummary } from "../interfaces/dns-hole-summary/dns-hole-summary-types";
+import { summaryResponseSchema } from "./pi-hole-types";
+
+export class PiHoleIntegration
+  extends Integration
+  implements DnsHoleSummaryIntegration
+{
+  async getSummaryAsync(): Promise<DnsHoleSummary> {
+    const apiKey = super.getSecretValue("apiKey");
+    const response = await fetch(
+      `${this.url}/admin/api.php?summaryRaw&auth=${apiKey}`,
+    );
+    if (!response.ok) {
+      throw new Error(`Failed to fetch summary: ${response.statusText}`);
+    }
+
+    const result = summaryResponseSchema.safeParse(await response.json());
+
+    if (!result.success) {
+      throw new Error(
+        `Failed to parse summary, most likely your api key is wrong: ${result.error.message}`,
```

Comment:
Add information about which integration is failing. 

Rephrase to not say that the api key is wrong, otherwise people will think that's the only thing that could go wrong. 

"[{integration}]: Failed to parse summary. Error message: {error}"

Reply:
But it would make sense to put them on the right track or not?